### ISSUE TITLE
[Failing test] 'should read runtime metrics' test is failing when NodeLocalDNS=true

### ISSUE THREAD

**Issue by ialidzhikov** [gardener/gardener#2932]
**How to categorize this issue?**

/area networking
/kind bug
/priority normal

**What happened**:
`should read runtime metrics` test seems to be failing when NodeLocalDNS feature gate is enabled.

**What you expected to happen**:
No such failure.

**How to reproduce it (as minimally and precisely as possible)**:
1. Start gardenlet with 

```yaml
apiVersion: gardenlet.config.gardener.cloud/v1alpha1
kind: GardenletConfiguration
# ...
featureGates:
  NodeLocalDNS: true
```

2. Create a new Shoot

3. Run the `should read runtime metrics` test against the newly created Shoot.

```
go test -timeout=0 -mod=vendor ./test/suites/shoot \
      --v -ginkgo.v -ginkgo.progress -ginkgo.noColor \
      --report-file=/tmp/report.json \
      -kubecfg=$KUBECONFIG \
      -shoot-name=$SHOOT_NAME \
      -project-namespace=$PROJECT_NAMESPACE \
      -ginkgo.focus="Shoot application metrics testing"
```

4. Ensure that the test is failing

```
• Failure [654.337 seconds]
Shoot application metrics testing
/Users/i331370/go/src/github.com/gardener/gardener/test/integration/shoots/applications/metrics.go:54

  /Users/i331370/go/src/github.com/gardener/gardener/test/framework/test_options.go:56
    [BETA] [SHOOT] should read runtime metrics [It]
    /Users/i331370/go/src/github.com/gardener/gardener/test/framework/gingko_utils.go:26

    Unexpected error:
        <*retry.retryError | 0xc000898000>: {
            ctxError: {},
            err: {
                ErrStatus: {
                    TypeMeta: {Kind: "", APIVersion: ""},
                    ListMeta: {
                        SelfLink: "",
                        ResourceVersion: "",
                        Continue: "",
                        RemainingItemCount: nil,
                    },
                    Status: "Failure",
                    Message: "podmetrics.metrics.k8s.io \"gardener-e2e-vxw4c/metrics-test-7f4d6df99b-rbwqp\" not found",
                    Reason: "NotFound",
                    Details: {
                        Name: "gardener-e2e-vxw4c/metrics-test-7f4d6df99b-rbwqp",
                        Group: "metrics.k8s.io",
                        Kind: "podmetrics",
                        UID: "",
                        Causes: nil,
                        RetryAfterSeconds: 0,
                    },
                    Code: 404,
                },
            },
        }
        retry failed with context deadline exceeded, last error: podmetrics.metrics.k8s.io "gardener-e2e-vxw4c/metrics-test-7f4d6df99b-rbwqp" not found
    occurred

    /Users/i331370/go/src/github.com/gardener/gardener/test/integration/shoots/applications/metrics.go:85
------------------------------


Summarizing 1 Failure:

[Fail] Shoot application metrics testing  [It] [BETA] [SHOOT] should read runtime metrics
/Users/i331370/go/src/github.com/gardener/gardener/test/integration/shoots/applications/metrics.go:85

Ran 1 of 19 Specs in 654.338 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 18 Skipped
--- FAIL: TestGardenerSuite (654.36s)
FAIL
FAIL	github.com/gardener/gardener/test/suites/shoot	655.369s
FAIL
```

Also with kubectl:

```
$ k -n kube-system top po
W1001 12:31:50.477301   37165 top_pod.go:274] Metrics not available for pod kube-system/calico-kube-controllers-68657c7d94-hwszr, age: 23m9.477288s
error: Metrics not available for pod kube-system/calico-kube-controllers-68657c7d94-hwszr, age: 23m9.477288s

$ k top no
error: metrics not available yet
```

Logs of metrics-server:

```
I1001 09:36:04.532056       1 log.go:172] http: TLS handshake error from 10.222.0.27:56384: EOF
E1001 09:36:10.285087       1 reststorage.go:160] unable to fetch pod metrics for pod kube-system/coredns-97dfb4f59-qlmhj: no metrics known for pod
E1001 09:36:10.285120       1 reststorage.go:160] unable to fetch pod metrics for pod kube-system/coredns-97dfb4f59-cp6nr: no metrics known for pod
I1001 09:36:11.924066       1 log.go:172] http: TLS handshake error from 10.222.0.27:56450: EOF
E1001 09:36:15.745889       1 reststorage.go:160] unable to fetch pod metrics for pod gardener-e2e-fhp5q/metrics-test-7f4d6df99b-979bw: no metrics known for pod
E1001 09:36:17.071619       1 reststorage.go:160] unable to fetch pod metrics for pod gardener-e2e-fhp5q/metrics-test-7f4d6df99b-979bw: no metrics known for pod
I1001 09:36:19.520117       1 manager.go:148] ScrapeMetrics: time: 30.007957004s, nodes: 0, pods: 0
E1001 09:36:19.520143       1 manager.go:111] unable to fully collect metrics: unable to fully scrape metrics from source kubelet_summary:ip-10-222-0-27.eu-west-1.compute.internal: unable to fetch metrics from Kubelet ip-10-222-0-27.eu-west-1.compute.internal (ip-10-222-0-27.eu-west-1.compute.internal): Get https://ip-10-222-0-27.eu-west-1.compute.internal:10250/stats/summary?only_cpu_and_memory=true: dial tcp: i/o timeout
I1001 09:36:21.924958       1 log.go:172] http: TLS handshake error from 10.222.0.27:56536: EOF
```

**Anything else we need to know?**:

**Environment**:

- Gardener version: master
- Kubernetes version (use `kubectl version`): v1.19.2
- Cloud provider or hardware configuration: AWS
- Others:

**Comment by rfranzke** [comment]
Is it related to https://github.com/gardener/gardener/issues/2745 ?

**Comment by ialidzhikov** [comment]
> Is it related to #2745 ?

It could be related. @DockToFuture , can you confirm?

**Comment by ialidzhikov** [comment]
ping @DockToFuture

**Comment by ialidzhikov** [comment]
It seems that the issue is fixed with #2745 (or any other PR meanwhile). I ran the test multiple times with the same setup with the current `gardener/gardener@master` and the test passed successfully.

/close
